api/v1beta1/openstackmachine_types.go

```diff
@@ -33,6 +33,41 @@ const (
 	IPClaimMachineFinalizer = "openstackmachine.infrastructure.cluster.x-k8s.io/ip-claim"
 )
 
+// SchedulerHintAdditionalValue represents the value of a scheduler hint property.
+// The value can be of various types: bool, string, int, or an object.
+// The Type field indicates the type of the value being used.
+type SchedulerHintAdditionalValue struct {
+	// Type represents the type of the value (e.g., "bool", "string", "int").
+	Type string `json:"type"`
+
+	// Bool is the boolean value of the scheduler hint, used when Type is "bool".
+	// This field is optional and will be omitted if not applicable.
```

Comment:
```suggestion
	// This field is required if type is 'Bool', and must not be set otherwise.
```

And similar for other fields.